packages/dx-grid-core/src/plugins/table-header-row/helpers.ts

```diff
@@ -67,11 +68,16 @@ export const getLastColumnName: LastColumnNameFn = (tableColumns) => {
     : undefined;
 };
 
-export const getNextColumnName: NextColumnNameFn = (tableColumns, columnName) => {
+const nextColumnName: NextColumnNameFn = (tableColumns , index) => {
+  const haveNextColumn = index >= 0 && index < tableColumns.length - 1;
```

Comment:
I think the next changes are more cleary

`haveNextColumn` -> `isNextColumnExists`

OR

```js
const nextColumn = index >= 0 && index < tableColumns.length - 1 && tableColumns[index + 1];
return nextColumn.type === TABLE_DATA_TYPE
  ? nextColumn.name
  : undefined;
```